### PR TITLE
ENT-3451 | added support for grade type report in enterprise report configurations.

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
related ticket: https://github.com/edx/edx-enterprise-data/pull/213/

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/models.py [hunk 1]
```diff
@@ -2037,12 +2037,14 @@ class EnterpriseCustomerReportingConfiguration(TimeStampedModel):
     DATA_TYPE_PROGRESS = 'progress'  # Refers to gathering progress data from Vertica (to be deprecated)
     DATA_TYPE_PROGRESS_V2 = 'progress_v2'  # Refers to gathering progress data from the Analytics Data API
     DATA_TYPE_CATALOG = 'catalog'
+    DATA_TYPE_GRADE = 'grade'
```
Comment (irfanuddinahmad) [inline enterprise/models.py:2040]: Can you add DATA_TYPE 'completion' also

Comment (HammadAhmadWaqas) [reply enterprise/models.py:2040]: yes, sure. we have to merge this PR after the edx-enterprise-data.
I will add three of them into a single PR.

**Review by zamanafzal — APPROVED** [review]

File: enterprise/admin/forms.py [hunk 2]
```diff
@@ -566,6 +566,18 @@ def clean(self):
         """
         cleaned_data = super(EnterpriseCustomerReportingConfigAdminForm, self).clean()
         report_customer = cleaned_data.get('enterprise_customer')
+        data_type = cleaned_data.get('data_type')
+
+        if data_type in EnterpriseCustomerReportingConfiguration.PEARSON_ONLY_REPORTS \
+                and report_customer.name != 'Pearson':
+            message = _(
+                'This data_type "{data_type}" is not supported for enterprise'
+                'customer {enterprise_customer}. please select a different data_type.',
```
Comment (irfanuddinahmad) [inline enterprise/admin/forms.py:575]: nit: please->Please

File: enterprise/migrations/0116_auto_20201116_0400.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ('enterprise', '0115_enterpriseanalyticsuser_historicalenterpriseanalyticsuser'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='enterprisecustomerreportingconfiguration',
+            name='data_type',
+            field=models.CharField(choices=[('progress', 'progress'), ('progress_v2', 'progress_v2'), ('catalog', 'catalog'), ('engagement', 'engagement'), ('grade', 'grade'), ('completion', 'completion'), ('course_structure', 'course_structure')], default='progress', help_text='The type of data this report should contain.', max_length=20, verbose_name='Data Type'),
```
Comment (irfanuddinahmad) [inline enterprise/migrations/0116_auto_20201116_0400.py:16]: Shouldn't course_structure be a single word?

Comment (HammadAhmadWaqas) [reply enterprise/migrations/0116_auto_20201116_0400.py:16]: I think course_structure is also fine. It's not causing any problem.

File: enterprise/admin/forms.py [hunk 4]
```diff
@@ -566,6 +566,18 @@ def clean(self):
         """
         cleaned_data = super(EnterpriseCustomerReportingConfigAdminForm, self).clean()
         report_customer = cleaned_data.get('enterprise_customer')
+        data_type = cleaned_data.get('data_type')
+
+        if data_type in EnterpriseCustomerReportingConfiguration.PEARSON_ONLY_REPORTS \
```
Comment (irfanuddinahmad) [inline enterprise/admin/forms.py:571]: please add a comment explaining that this code will be removed after the partial rollout.

**Review by irfanuddinahmad — APPROVED** [review]

**Pull request merged**